multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/AbstractKeycloakFilter.java

```diff
@@ -60,4 +84,57 @@ public void init(FilterConfig filterConfig) throws ServletException {}
 
   @Override
   public void destroy() {}
+
+  private class SigningKeyResolverAdapterByKind extends SigningKeyResolverAdapter {
+
+    @Override
+    public Key resolveSigningKey(JwsHeader header, String plaintext) {
+      if (MACHINE_TOKEN_KIND.equals(header.get("kind"))) {
+        throw new JwtException("Not a keycloak token");
```

Comment:
Is it really correct message or maybe not `MACHINE_TOKEN_KIND` should be used here?

Reply:
Correct

Reply:
The message is correct and `MACHINE_TOKEN_KIND` should be used while comparing a kind header?

Reply:
Yes. if it's machine token than it's not a keycloak token.

Reply:
Thanks, now it's clear. It would be useful to have a comment there that token which is not a machine is considered as keycloak because it is not really clear without context.